### PR TITLE
Remove Rails 3 attr_accessible from models

### PR DESCRIPTION
**Pull request by Bodacious**
Changes proposed in this PR:
- Removes deprecated `attr_accessible` from models
- Adds Rails 4 strong parameters to controllers
- Cleans up tests that broke after changing attr assignation in Plans
- Refactors some tests for readability
- Fixes bugs in `Section#deep_copy` and `Phase#deep_copy` where original record ID was being passed to children.

### CODE REVIEW

File: app/models/section.rb [hunk 1]
```diff
@@ -42,7 +38,7 @@ def deep_copy(**options)
     copy.modifiable = options.fetch(:modifiable, self.modifiable)
     copy.phase_id = options.fetch(:phase_id, nil)
     copy.save!(validate: false)  if options.fetch(:save, false)
-    options[:section_id] = id
+    options[:section_id] = copy.id
```
Comment (Bodacious) [inline app/models/section.rb]: @briri @xsrust Please have a look at this suspected bug and confirm. Thanks.

File: app/models/phase.rb [hunk 2]
```diff
@@ -37,7 +32,7 @@ def deep_copy(**options)
     copy.modifiable = options.fetch(:modifiable, self.modifiable)
     copy.template_id = options.fetch(:template_id, nil)
     copy.save!(validate:false)  if options.fetch(:save, false)
-    options[:phase_id] = id
+    options[:phase_id] = copy.id
```
Comment (Bodacious) [inline app/models/phase.rb]: @xsrust @briri Please have a look at this suspected bug and confirm.

**Comment by briri** [comment]
this is related to #100

**Comment by Bodacious** [comment]
Rebased on `development` branch to fix merge conflicts.

**Comment by briri** [comment]
@xsrust you wrote these `deep_copy` methods. If the proposed change looks ok to you too, please go ahead and merge :)

File: app/models/section.rb (hunk 1 above)
Comment (xsrust) [reply app/models/section.rb]: I'm also concerned about bad data and will check our services to see if anything needs cleaned up.

**Comment by briri** [comment]
@stephaniesimms and @sjDCC a bug was discovered in the template/phase/section copy and versioning methods. Please verify that the Template copy and versioning are still working properly

**Comment by sjDCC** [comment]
I'm getting 404 errors when I create a new template at the phase creation point, or when I copy a template at the phase editing point. Can't get past those to test other actions

**Comment by briri** [comment]
Likely due to my bringing the system offline a few minutes ago. Should be ok now @sjDCC  and @stephaniesimms

**Comment by sjDCC** [comment]
I'm still getting errors so suspect it's a bug

capture

**Comment by sjDCC** [comment]
We seem to still have problems on template copying and editing so not sure this is fixed.

If I click the 'Actions' dropdown and select copy template I get this error
copy-template

It seems like editing an existing template works, but when I tried to add a new question I got this error.
new-version

The customise template actions are still throwing a section creation error too but perhaps the latest changes here haven't been merged and deployed?
section-creation

**Comment by sjDCC** [comment]
I've just tested this on DMPonline-dev and can copy templates now and create custom sections. The middle issue above persists though - when you add or edit an existing question in a custom section it throws an error "Unable to create a new version of this template." Perhaps a permissions thing?

**Comment by sjDCC** [comment]
This ticket is probably fine to close out now. The only remaining issue (the error thrown when adding/editing questions in custom sections) seems to be a wider problem which has been raised separately in #1817

**Pull request merged**